src/mem/metaslab.h

```diff
@@ -102,10 +102,13 @@ namespace snmalloc
     {
       size_t size = sizeclass_to_size(sizeclass);
       size_t offset = get_slab_offset(sizeclass, is_short);
+      size_t all_ones = std::numeric_limits<size_t>::is_signed ?
+        size_t(-1) :
+        std::numeric_limits<size_t>::max();
 
       size_t head_start =
-        remove_cache_friendly_offset(head & ~static_cast<size_t>(1), sizeclass);
-      size_t slab_start = offset & ~static_cast<size_t>(1);
+        remove_cache_friendly_offset(head & all_ones, sizeclass);
```

Comment:
What is this doing?
```C 
& all_ones
```
looks like a nop here.  Has something gotten broken here?

I think this use to be
```C
      size_t head_start =
        remove_cache_friendly_offset(head & ~(size_t)1, sizeclass);
      size_t slab_start = offset & ~(size_t)1;
```

Looks like it is unsetting the bottom bit. I.e. `_ & 0xffff...fffe`

Reply:
Ah, this was me not reading the code that I was changing properly.